detail.py

```diff
@@ -0,0 +1,25 @@
+from django import forms
+from django.http import HttpResponseRedirect
+from django_mako_plus import view_function, jscontext, RedirectException
+from catalog import models as cmod
+from formlib import Formless
+import math
+
+@view_function
+def process_request(request, product:cmod.Product):
+
+    if product in request.last_five:
+        request.last_five.remove(product)
```

Comment:
You could use .pop instead of .remove and then you can pop something off the list and add something to a list at the same time.

Reply:
I feel remove is the easiest way because pop will always pop the last one.